packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -346,18 +345,32 @@ class TreeViewExtImpl<T> implements Disposable {
         return idLabel;
     }
 
-    private buildTreeItemId(parentId: string, item: TreeItem): string {
-        // build tree id according to https://code.visualstudio.com/api/references/vscode-api#TreeItem
-        // note: the front end tree implementation cannot handle reparenting items, hence the id is set to the "path" of individual ids
-        let id = typeof item.id === 'string' ? item.id : this.getItemLabel(item);
-        if (id) {
-            // we use '' as the id of the root, we don't consider that a valid id
-            // since '' is falsy, we'll never get '' in this branch
-            id = TreeViewExtImpl.ID_ITEM + id;
-        } else {
-            id = TreeViewExtImpl.ID_COMPUTED + this.nextItemId++;
+    // Modeled on https://github.com/microsoft/vscode/blob/main/src/vs/workbench/api/common/extHostTreeViews.ts#L822
+    private buildTreeItemId(parentId: string, item: TreeItem, mustReturnNew: boolean): string {
```

Comment:
I'm not sure what `mustReturnNew` means and why it's true in one case, and false in the other.

Reply:
But why would it ever makes sense to do `mustReturnNew = false`? for multiple elements that have the same computed "id", it will just return the first one, no? Would that be correct?

Reply:
Yes, I think that's a bit odd as well, since it seems non-deterministic, but the same is true in VSCode - in the parallel calls, the corresponding flag is true in one place and false in the other. Given the ID's are being computed, I suppose the idea is that really can't guarantee that we'll find the same one, so we'll take the first one that's as close as we can reasonably get, given the data available.